machine/quanta/quanta_lb9/u-boot/Fixup-MPC8541-SS_EN-and-CLK_ADJ.patch

```diff
@@ -0,0 +1,52 @@
+From c2cbc2ba4699f1b529e16a98ddfa89d98ee89bc4 Mon Sep 17 00:00:00 2001
+From: Luffy <luffy.cheng@quantatw.com>
+Date: Mon, 9 Mar 2015 14:14:14 +0800
+Subject: [PATCH 1/1] Fixup MPC8541 SS_EN and CLK_ADJ
+
+---
+ arch/powerpc/cpu/mpc8xxx/ddr/ctrl_regs.c |    4 ++++
+ board/quanta/quanta_lb9/ddr.c            |   11 ++++++++---
+ 2 files changed, 12 insertions(+), 3 deletions(-)
+
+diff --git a/arch/powerpc/cpu/mpc8xxx/ddr/ctrl_regs.c b/arch/powerpc/cpu/mpc8xxx/ddr/ctrl_regs.c
+index 6265e52..2c71089 100644
+--- a/arch/powerpc/cpu/mpc8xxx/ddr/ctrl_regs.c
++++ b/arch/powerpc/cpu/mpc8xxx/ddr/ctrl_regs.c
+@@ -1215,7 +1215,11 @@ static void set_ddr_sdram_clk_cntl(fsl_ddr_cfg_regs_t *ddr,
+ 	unsigned int clk_adjust;	/* Clock adjust */
+ 
+ 	clk_adjust = popts->clk_adjust;
++#if defined(CONFIG_MPC8541) || defined(CONFIG_MPC8555)
++	ddr->ddr_sdram_clk_cntl = (0x80000000 | ((clk_adjust & 0x7) << 24));
++#else
+ 	ddr->ddr_sdram_clk_cntl = (clk_adjust & 0xF) << 23;
++#endif
+ 	debug("FSLDDR: clk_cntl = 0x%08x\n", ddr->ddr_sdram_clk_cntl);
+ }
```

Comment:
I was wondering if SS_EN patch to a global core ONIE, impacts to all user who are using MPC8555/8541, they need to modify correct clk_adjust as 24 bits shift, otherwise DDR config may cause unknow problem.


From: Curt Brune [mailto:notifications@github.com]
Sent: Wednesday, March 11, 2015 11:58 PM
To: opencomputeproject/onie
Cc: *QuantaSwitchONIE
Subject: Re: [onie] Quanta LB9: Fixup MPC8541 SS_EN and CLK_ADJ (#149)


In machine/quanta/quanta_lb9/u-boot/Fixup-MPC8541-SS_EN-and-CLK_ADJ.patch<https://github.com/opencomputeproject/onie/pull/149#discussion_r26225419>:

> +

> +diff --git a/arch/powerpc/cpu/mpc8xxx/ddr/ctrl_regs.c b/arch/powerpc/cpu/mpc8xxx/ddr/ctrl_regs.c

> +index 6265e52..2c71089 100644

> +--- a/arch/powerpc/cpu/mpc8xxx/ddr/ctrl_regs.c

> ++++ b/arch/powerpc/cpu/mpc8xxx/ddr/ctrl_regs.c

> +@@ -1215,7 +1215,11 @@ static void set_ddr_sdram_clk_cntl(fsl_ddr_cfg_regs_t *ddr,

> +     unsigned int clk_adjust;       /* Clock adjust */

> +

> +     clk_adjust = popts->clk_adjust;

> ++#if defined(CONFIG_MPC8541) || defined(CONFIG_MPC8555)

> ++    ddr->ddr_sdram_clk_cntl = (0x80000000 | ((clk_adjust & 0x7) << 24));

> ++#else

> +     ddr->ddr_sdram_clk_cntl = (clk_adjust & 0xF) << 23;

> ++#endif

> +     debug("FSLDDR: clk_cntl = 0x%08x\n", ddr->ddr_sdram_clk_cntl);

> + }

This looks good, but this should be two separate patches.

The SS_EN fix is generic to all MPC8541 and MPC8555, while the below is specific to the LB9.

The SS_EN fix should be a core ONIE patch with the prefix "arch-powerpc-XXX" in this directory https://github.com/opencomputeproject/onie/tree/master/patches/u-boot.

—
Reply to this email directly or view it on GitHub<https://github.com/opencomputeproject/onie/pull/149/files#r26225419>.